lib/commands/debug/index.js

```diff
@@ -0,0 +1,200 @@
+                debugRegion, (err, debugFlow) => {
+                    if (err) {
+                        Messenger.getInstance().error(err);
+                        return cb(err);
+                    }
+                    if (cmd.waitForAttach) {
+                        Messenger.getInstance().info(`Debugging session will wait until inspector is attached at port - ${debugPort}`);
+                    }
+                    debugFlow.execCommand();
+                });
+        });
+    }
+
+    _getAccessTokenForProfile(profile, debug, callback) {
+        const authorizationController = new AuthorizationController({
+            auth_client_type: 'LWA',
+            doDebug: debug
+        });
+        authorizationController.tokenRefreshAndRead(profile, (tokenErr, token) => {
+            if (tokenErr) {
+                return callback(tokenErr);
+            }
+            callback(null, token);
+        });
+    }
+
+    _getHostedSkillRuntime(smapiClient, skillId, callback) {
+        smapiClient.skill.alexaHosted.getAlexaHostedSkillMetadata(skillId, (err, response) => {
+            if (err) {
+                return callback(err);
+            }
+            if (response.statusCode >= 300) {
+                const error = jsonView.toString(response.body);
+                return callback(error);
+            }
+            const { runtime } = response.body.alexaHosted;
+            if (!stringUtils.isNonBlankString(runtime)) {
+                return callback(`Unable to determine runtime of the hosted skill - ${skillId}`);
+            }
+            callback(null, runtime);
+        });
+    }
+
+    _getSkillDebugflow(skillId, profile, skillCodeRegion, waitForAttach, debug, debugPort, token, debugRegion, callback) {
+        let DebugFlow;
+        if (this._filterAlexaHostedSkill(profile)) {
+            const smapiClient = new SmapiClient({
+                profile,
+                doDebug: debug
+            });
+            this._getHostedSkillRuntime(smapiClient, skillId, (err, runtime) => {
+                if (err) {
+                    return callback(err);
+                }
+                DebugFlow = this._selectDebugFlowClass(runtime);
+                try {
+                    const debugFlowInstance = new DebugFlow({
+                        skillInvocationInfo: helper.getHostedSkillInvocationInfo(runtime),
+                        waitForAttach,
+                        debugPort,
+                        token,
+                        skillId,
+                        debugRegion
+                    });
+                    callback(null, debugFlowInstance);
+                } catch (error) {
+                    callback(error);
+                }
+            });
+        } else {
+            const skillCodeFolderName = ResourcesConfig.getInstance().getCodeSrcByRegion(profile, skillCodeRegion);
+            if (!stringUtils.isNonBlankString(skillCodeFolderName)) {
```

Comment:
I am still not 100% about all the validations here. there are not debug command specific. I feel like we should have some common layer for validations. @RonWang what do you think?

Reply:
It's relative out of the focus of this cr, and in my mind we cannot have the validation together with resourcesConfig directly, since different caller has different need: some might need the getFilePath fail when it's not existing, some might want to create the file when getFilePath fail. We cannot unify the validation easily. The only thing we can do is to create some general validation, like create a class as FilePath then load the value from ResourcesConfig.getXxxPath